### PR TITLE
gh-103822: change return value to enum for day and month APIs

### PR DESCRIPTION
**Pull request by Agent-Hellboy**
this is an initial commit to get some quick feedback on calendar APIs which I can change 

* Issue: gh-103822

### CODE REVIEW

File: Lib/calendar.py [hunk 1]
```diff
@@ -70,6 +70,13 @@ class Day(IntEnum):
     SUNDAY = 6
 
 
+def _get_enum_attribute(value, enum_type):
```
Comment (ethanfurman) [inline Lib/calendar.py]: This isn't necessary.

File: Lib/calendar.py [hunk 2]
```diff
@@ -143,7 +150,7 @@ def weekday(year, month, day):
     """Return weekday (0-6 ~ Mon-Sun) for year, month (1-12), day (1-31)."""
     if not datetime.MINYEAR <= year <= datetime.MAXYEAR:
         year = 2000 + year % 400
-    return datetime.date(year, month, day).weekday()
+    return _get_enum_attribute(datetime.date(year, month, day).weekday(),'day')
```
Comment (ethanfurman) [inline Lib/calendar.py]:
```suggestion
    return Day(datetime.date(year, month, day).weekday())
```

**Review by ethanfurman — CHANGES_REQUESTED** [review]
I thought there would be more places in the file where `Day` and `Month` would be used.  Oh, well.

Make those two changes and we'll call it good.

**Comment by Agent-Hellboy** [comment]
I have made the requested changes; please review again

**Review by ethanfurman — APPROVED** [review]

**Pull request merged**